Added flags to the frontend test runner script

BenHenning opened the pull request:
Updated the frontend test script to not run the second pass of tests by default. It still runs for the presubmit check, but not by default. Also, added an optional flag to skip the installation process. This is useful for when third party dependencies have already been installed. These two flags together significantly speed up running the frontend tests. We may want to consider setting the install dependencies flag on by default and having the frontend checks error-out if third party dependencies haven't been installed yet.

seanlip commented on scripts/run_frontend_tests.sh:
```diff
@@ -37,28 +37,60 @@ set -e
-echo ""
-echo "  Running build task with concatenation only "
-echo ""
+# Parse additional command line arguments.
```
We use a similar pattern in run_e2e_tests.sh. Can you abstract both instances out into a common function (maybe in setup.sh) to avoid duplication?

BenHenning replied:
Done. This might slightly impact the performance of running the frontend tests before dependencies are installed (since it will also now install the protractor dependencies for both, ditto for end-to-end tests), but it's inconsequential.

seanlip commented on scripts/run_frontend_tests.sh:
```diff
@@ -37,28 +37,60 @@ set -e
-echo ""
+# Parse additional command line arguments.
+# Credit: http://stackoverflow.com/questions/192249
+SKIP_INSTALL=false
```
Suggest SKIP_INSTALLING_THIRD_PARTY_LIBS

('install' seems too general -- to me, it refers to setting up Oppia itself as well)

seanlip replied:
(The arg typed by the developer can still be `skip-install`)

BenHenning replied:
Sgtm, done.

seanlip commented:
Thanks, Ben! Assigning to @bbriggs and @MAKOSCAFEE for a functionality review; I also made some comments on the code.

MAKOSCAFEE commented:
Hi @BenHenning  I tried to run front-end test with the flag `bash scripts/run_frontend_tests.sh --run_minified_tests=true` but I could not see it run in production environment. 
Other thing is I think as how you added the flag in pre-submit script the same could be done in travis., what do you think?

seanlip commented:
@MAKOSCAFEE could you please elaborate more on "the same could be done in travis"? Not sure I understand (so just thought I'd check).

MAKOSCAFEE commented:
so for now if you just run `bash scripts/run_frontend_tests.sh` this will run front-end tests with concatenation only(no minification) which is ok for dev environment and this is what is run in travis. But since @BenHenning  added  `--run_minified_tests=true` to make it run using minified files(css and js) I think It would be better to add the same flag in travis as it  was done in [pre-submit script.](https://github.com/oppia/oppia/pull/1588/files#diff-d1c6cf3975a82e5dc0ed4378f96ffbc5R52)

seanlip commented:
Ah yes, thanks for explaining. I agree with that!

BenHenning commented:
Thanks @seanlip, @MAKOSCAFEE! I think I addressed all review comments. @MAKOSCAFEE good call on ensuring Travis needing to run both sets of the frontend tests. Regarding the production bit, the usage is as follows:

``bash scripts/run_frontend_tests.sh --run-minified-tests=true``

I made a typo in the presubmit script, which I've fixed. I also added documentation to ``run_frontend_tests.sh`` and ``run_e2e_tests.sh`` to ensure others don't make the same mistake when using the scripts in the future.

bbriggs commented:
Are we using artifacts at all in travis tests?

bbriggs commented on scripts/run_e2e_tests.sh:
```diff
@@ -22,6 +22,7 @@
 #   bash scripts/run_e2e_tests.sh
 #
 # Optional arguments:
+#   --skip-install=true/false Possibly skips installing dependencies. The default value is false.
```
In what way does it "possibly" skip installing deps?

BenHenning replied:
I clarified in the comment, PTAL.

bbriggs commented on scripts/setup.sh:
```diff
@@ -18,6 +18,71 @@
+      ;;
+
+      *)
+      echo Error: Unknown command line option: $i
```
We should probably wrap this string in quotes and add a brace around `i`

BenHenning replied:
Done (and elsewhere).

BenHenning commented:
Thanks @bbriggs. I addressed your review comments.

bbriggs commented:
I think everyone else has put their LGTM on it. Thanks for the updates. Looks good!
